### PR TITLE
Feature/efk

### PR DESCRIPTION
**Pull request by akastav**

### CODE REVIEW

File: modules/logging/efk/README.md [hunk 1]
```diff
@@ -1,3 +1,35 @@
-This chart installs an elasticsearch cluster with kibana, elastic-curator and filebeat by default. You can optionally enable logstash and install Fluentd if you prefer. It also optionally installs nginx-ldapauth-proxy.
+## Elastic Kibana Filebeat
 
-[Elastick-stack Documentation](https://github.com/helm/charts/tree/master/stable/elastic-stack)
+https://helm.elastic.co/
+
+
+## Prerequisite
```
Comment (RustamGimadiev) [inline modules/logging/efk/README.md:6]: is it possible to use this module without Google OAuth keys?

File: modules/logging/efk/main.tf [hunk 2]
```diff
@@ -1,75 +1,293 @@
...
+  filebeat_conf       = merge(local.filebeat_conf_defaults, var.filebeat_conf)
+  elastic_conf_defaults = {
+    "replicas" = var.elasticReplicas
+    "minimumMasterNodes" = var.elasticMinMasters
+    "volumeClaimTemplate.resources.requests.storage" = var.elasticDataSize
+  }
+  kibana_conf_defaults = {
+    "elasticsearchHosts" = "http://elasticsearch-master:9200"
+    "ingress.enabled"                                                   = "true"
+    "ingress.hosts[0]"                                                  = "kibana.${var.domains[0]}"
```
Comment (RustamGimadiev) [inline modules/logging/efk/main.tf:168]: in code bellow you define `var.domains` as a string variable 
``` hcl
variable "domains" {
  description = "domain name for ingress"
  default     = "example.com"
}
```
but use it as a non-zero length list of strings, maybe better to wrap all these options to the loop over `var.domains`?

File: modules/logging/efk/main.tf [hunk 3]
```diff
@@ -1,75 +1,293 @@
...
+    "ingress.annotations.kubernetes\\.io/ingress\\.class"               = "nginx"
+    "ingress.annotations.nginx\\.ingress\\.kubernetes\\.io/auth-url"    = "https://${var.efk_oauth2_domain}.${var.domains[0]}/oauth2/auth"
```
Comment (RustamGimadiev) [inline modules/logging/efk/main.tf:170]: imho, better to leave it to the user to define the ingress class and security annotations for incoming traffic

Comment (RustamGimadiev) [reply modules/logging/efk/main.tf:170]: in addition, seems like we should add `if` condition here to wrap situations when oauth2_enabled is `false`

File: modules/logging/efk/README.md [hunk 4]
```diff
@@ -1,3 +1,35 @@
...
+```
+module "efk" {
+  module_depends_on = [module.argocd.state.path]
+  source            = "../modules/logging/efk"
+  cluster_name      = module.kubernetes.cluster_name
+  argocd            = module.argocd.state
+  efk_oauth2_domain = "auth"
+  domains           = local.domain
+}
+```
```
Comment (RustamGimadiev) [inline modules/logging/efk/README.md:35]: can you please add a table of input parameters and outputs, you can simply generate it by `terraform-docs markdown .` and past it here as is

**Review by RustamGimadiev — CHANGES_REQUESTED** [review]

**Review by gfrntz — COMMENTED** [review]
Could you update common/examples/main.tf with complete example?

File: modules/logging/efk/README.md (hunk 1 above)
Comment (akastav) [reply modules/logging/efk/README.md:6]: I fixed it. If we don't set efk_oauth_domain - deploy kibana without google auth

File: modules/logging/efk/main.tf (hunk 3 above)
Comment (akastav) [reply modules/logging/efk/main.tf:170]: fixed

File: modules/logging/efk/variables.tf [hunk 5]
```diff
@@ -1,52 +1,98 @@
...
+  type        = string
+  description = "Number of minimum elasticsearch master nodes. Keep this number low or equals that Replicas"
+  default     = "2"
+}
+
+variable "elasticDataSize" {
+  type        = string
+  description = "Request pvc size for elastic volume data size"
+  default     = "30Gi"
+}
```
[the remark this answers is not in the document]
Comment (akastav) [reply modules/logging/efk/variables.tf:98]: updated

**Pull request closed without merging**